modules/account-lib/src/coin/algo/keyRegistrationBuilder.ts

```diff
@@ -149,11 +162,13 @@ export class KeyRegistrationBuilder extends TransactionBuilder {
 
     if (algoTxn) {
       if (!algoTxn.nonParticipation) {
-        this.voteKey(algoTxn.voteKey.toString('base64'));
-        this.selectionKey(algoTxn.selectionKey.toString('base64'));
-        this.voteFirst(algoTxn.voteFirst);
-        this.voteLast(algoTxn.voteLast);
-        this.voteKeyDilution(algoTxn.voteKeyDilution);
+        if (algoTxn.voteKey && algoTxn.selectionKey && algoTxn.voteFirst && algoTxn.voteLast && algoTxn.voteKeyDilution) {
```

Comment:
Seems this code is duplicated in several places (the condition), can you extract it to a function?